cmd/query/app/server.go

```diff
@@ -49,6 +49,9 @@ import (
 
 // Server runs HTTP, Mux and a grpc server
 type Server struct {
+	mu           sync.Mutex
+	started      bool
+	stopChan     chan struct{}
```

Comment:
Why do we need any of these? If you suspect that Close is being called twice from the test we can just add a log message. But there's no evidence it was called twice. And the channel is never read at all.

Reply:
Thank you for the insights. The additional fields were intended to prevent redundant close operations, but I understand that there's no current evidence of the Close method being called twice. I'll simplify the code by removing the `stopChan` and focus on logging to diagnose the issue more clearly.